src/lib/containers/drug_upload_tool/DrugUploadForm.tsx

```diff
@@ -142,9 +139,15 @@ export default class DrugUploadForm extends React.Component<
 
   componentDidUpdate(prevProps: DrugUploadFormProps) {
     const shouldUpdate = this.props.callbackStatus !== prevProps.callbackStatus
-    const isSuccess = this.props.callbackStatus === StatusEnum.SAVE_SUCCESS  || this.props.callbackStatus === StatusEnum.SUBMIT_SUCCESS
+    const isSuccess =
+      this.props.callbackStatus === StatusEnum.SAVE_SUCCESS ||
+      this.props.callbackStatus === StatusEnum.SUBMIT_SUCCESS
     if (shouldUpdate && isSuccess) {
       this.setState({ hasUnsavedChanges: false })
+      if (this.props.callbackStatus === StatusEnum.SUBMIT_SUCCESS) {
+        this.setState({ isSubmitted: true })
+        window.history.back()
```

Comment:
This will usually work, and it's understandable why you did it this way (since the component has no idea what path corresponds to the grid)